src/storage/phonesContext.tsx

```diff
@@ -0,0 +1,245 @@
+import React, {
+  Dispatch,
+  SetStateAction,
+  useState,
+} from 'react';
+import { Product } from '../types/Product';
+import { SortType } from '../types/SortType';
+import { SortParamsType } from '../types/SortParamsType';
+import { useLocalStorage } from '../hooks/useLocalStorage';
+import { CartProduct } from '../types/CartProduct';
+
+type PhonesContextType = {
+  products: Product[],
+  setProducts: Dispatch<SetStateAction<Product[]>>,
+  preparedBrandNewProducts: Product[],
+  preparedHotPriceProducts: Product[],
+  sortType: SortType,
+  setSortType: Dispatch<SetStateAction<SortType>>,
+  itemsPerPage: number,
+  setItemsPerPage: Dispatch<SetStateAction<number>>,
+  sortParams: SortParamsType[],
+  perPageParams: number[],
+  sortedProducts: () => Product[],
+  tabletSearchValue: string,
+  setTabletSearchValue: Dispatch<SetStateAction<string>>,
```

Comment:
it's better to use search params to store `searchValue`
No need to save it when you change the tab

same for all sort params